lib/LightPanels/LightPanels.js

```diff
@@ -524,29 +548,22 @@ module.exports = kind(
 	* @private
 	*/
 	transitionFinished: function (sender, ev) {
-		if (ev.originator === this._currentPanel) {
-			if ((this._indexDirection < 0 && this.popOnBack && this.index < this.getPanels().length - 1) ||
-				(this._indexDirection > 0 && this.popOnForward && this.index > 0)) {
-				this.popPanels(this.index, this._indexDirection);
-			}
-			if (this._currentPanel.postTransition) {
-				asyncMethod(this, function () {
-					this._currentPanel.postTransition();
-				});
-			}
-			if (this._garbagePanels && this._garbagePanels.length) {
-				this.finalizePurge();
-			}
+		var panel = ev.originator;
 
-			if (this._currentPanel) {
-				this._currentPanel.removeClass('transitioning');
-				this._currentPanel.activated && this._currentPanel.activated();
+		if (panel === this._currentPanel) {
+			if ((this._indexDirection < 0 && (this.popOnBack || this.cacheViews) && this.index < this.getPanels().length - 1) ||
+				(this._indexDirection > 0 && (this.popOnForward || this.cacheViews) && this.index > 0)) {
+				this.popPanels(this.index, this._indexDirection);
 			}
+			if (this.popQueue && this.popQueue.length) this.finalizePurge();
+			this.transitioning = false;
+		}
 
-			if (this._previousPanel) {
-				this._previousPanel.removeClass('transitioning');
-				this._previousPanel.deactivated && this._previousPanel.deactivated();
-			}
+		panel.removeClass('transitioning');
+		if (panel.postTransition) {
```

Comment:
These checks aren't necessary since enyo/LightPanel contains a no-op implementation of the method.